source/distributed/tria.cc

```diff
@@ -3862,6 +3912,59 @@ namespace parallel
           AssertThrow (false, ExcInternalError());
         }
 
+#ifdef DEBUG
+      // Check that we know the level subdomain ids of all our neighbors. This
+      // also involves coarser cells that share a vertex if they are active.
+      //
+      // Example (M= my, O=other):
+      //         *------*
+      //         |      |
+      //         |  O   |
+      //         |      |
+      // *---*---*------*
+      // | M | M |
+      // *---*---*
+      // |   | M |
+      // *---*---*
+      //  ^- the parent can be owned by somebody else, so O is not a neighbor
+      // one level coarser
+      if (settings & construct_multigrid_hierarchy)
+        {
+          for (unsigned int lvl=0; lvl<this->n_global_levels(); ++lvl)
+            {
+              std::vector<bool> active_verts = this->mark_locally_active_vertices_on_level(lvl);
+
+              typename Triangulation<dim, spacedim>::cell_iterator cell = this->begin(lvl>0?(lvl-1):lvl),
+                                                                   endc = this->end(lvl);
+              for (; cell != endc; ++cell)
+                if (cell->level() == lvl || cell->active())
+                  {
+                    bool is_level_artificial = cell->level_subdomain_id() == numbers::artificial_subdomain_id;
```

Comment:
mark `const`? (Also, maybe put parentheses around the initializer to make the grouping easier to read.)